src/ngraph_builder.cc

```diff
@@ -3658,6 +3657,11 @@ static Status TranslateStridedSliceOp(
   auto dim_vec = ng_input->get_shape();
   auto in_rank = dim_vec.size();
 
+  if (in_rank == 0) {
+    return errors::InvalidArgument("Index out of range using input dim ",
+                                   "; input has only ", in_rank, " dims");
+  }
```

Comment:
New tests:
x of dim [4,0,5].
`x[1:2, 2:2, 1:2]`. (slicing an empty dim by empty slice)
`x[1:2, 1:2, 1:2]`. (slicing an empty dim by non empty slice)

`x = 0` (scalar)
`x[:]` (supposed to fail.. this PR fixes the output error for this case)
`x[1:1]` (empty slice. Does this fail, if so what failure)

Reply:
@sayantan-nervana Added these tests, please review.